packages/table/src/TableHead/TableHead.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { PropsWithChildren } from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { stickyStyles, themeStyles } from './TableHead.styles';
+import { TableHeadProps } from './TableHead.types';
+
+const TableHead = ({
```

Comment:
Yeah that's an interesting point. It could feel redundant, but I think there are a few reasons why we might want this.

The first is to replicate how HTML markup works. You'd have
```
<thead>
    <tr>...</tr>
</thead>
``` 
in HTML, so it may be redundant markup, but it almost feels more intuitive to have parity to me and it would probably be the same for anyone that's written pure HTML tables.

The other is that although this feature is not currently designed/supported by us, both HTML tables and React-Table support [column groups](https://tanstack.com/table/v8/docs/examples/react/column-groups). In this case, it would be useful to not restrict the thead to just having one row.